src/crypto.c

```diff
@@ -0,0 +1,27 @@
+#include <aes_gcm.h>
+#include <sha256.h>
+#include <monocypher-ed25519.h>
+
+#include "crypto.h"
+
+fido_aes_gcm_encrypt_t fido_aes_gcm_encrypt = &aes_gcm_ae;
+
+fido_aes_gcm_decrypt_t fido_aes_gcm_decrypt = &aes_gcm_ad;
+
+void crypto_ed25519_sign_wrapper(uint8_t *signature,
+                                 const uint8_t *secret_key,
+                                 const uint8_t *message, size_t message_len) {
+    crypto_ed25519_sign(signature, secret_key, NULL, message, (int) message_len);
+                                
+                                 }
+fido_ed25519_sign_t fido_ed25519_sign = &crypto_ed25519_sign_wrapper;
+
+fido_ed25519_verify_t fido_ed25519_verify = &crypto_ed25519_check;
+
+fido_sha256_t fido_sha256 = &sha256;
```

Comment:
Pretty cool!

A possibility to make this configurable would be to have a fido_init function that receives a crypto descriptor and then sets this up. If the descriptor is NULL, the software crypto is used.

Reply:
My idea was to let users customize crypto functions by setting these function pointers from their code by including `crypto.h` in `fido.h`.

Reply:
Hm ok. Not that user friendly (because there is no documentation / wrapper that shows how to do this) but certainly a solution.

Reply:
Another idea that came to my mind was to additionally have some preprocessor flags whether to include a specific software-crypto-implementation.

Reply:
Ok.